pandas/core/indexes/category.py

```diff
@@ -328,23 +327,8 @@ def __contains__(self, key):
         if isna(key):  # if key is a NaN, check if any NaN is in self.
             return self.hasnans
 
-        # is key in self.categories? Then get its location.
```

Comment:
why can't you just do
```return key in self._data``?

Reply:
Ok, actually I'v'e got an good idea for this, will reimplement.